index.js

```diff
@@ -124,7 +125,25 @@ module.exports = (Hapi, options, allDone) => {
         value._name = key;
         pluginArr.push(value);
       });
-      pluginArr = _.sortBy(pluginArr, '_priority');
+      // use deptree to resolve the order in which the plugins load
+      // based on their _dependencies:
+      const deptree = new Deptree();
+      pluginArr.forEach((plugin) => {
+        // priority is deprecated:
+        if (plugin._priority) {
+          server.log(['error', 'hapi-confi'], `WARNING: '_priority' field is deprecated, please migrate your plugins to use the _dependencies format`);
+        }
+        deptree.add(plugin._name, plugin._dependencies ? plugin._dependencies : []);
+      });
+      pluginArr = _.reduce(deptree.resolve(), (memo, pluginName) => {
```

Comment:
do we need to do this reduce?  couldn't we just take the deptree.resolve() and pass to eachSeries?